### PR TITLE
pkg/asset/installconfig/ssh: Add OPENSHIFT_INSTALL_SSH_PUB_KEY_PATH

### PR DESCRIPTION
**Pull request by wking**
Builds on #350; review that first.

To make it easier to pass this in when you don't have a shell for:

```console
$ OPENSHIFT_INSTALL_SSH_PUB_KEY="$(cat path/to/key)" openshift-install install-config
```

For example, this is useful in CI where we can launch the cluster without hitting a shell at all.  And it's also a bit more compact for folks who are reading from files anyway.

I've also added a check for "do we only have one choice?".  If so, I just pick that value instead of bothering the user when they don't have a decision to make.

And I've reformatted the docs to use a list.  Tables are annoying to maintain in ASCII, and lists give us more space to talk about the expected input format, etc.

/assign @crawford

### CODE REVIEW

File: pkg/asset/installconfig/ssh.go [hunk 1]
```diff
@@ -39,32 +53,39 @@ func (a *sshPublicKey) Generate(map[asset.Asset]*asset.State) (state *asset.Stat
...
+					continue
+				}
+				pubKeys[path] = key
 			}
-
-			pubKeys[path] = pubKeyBytes
 		}
 	}
 
+	if len(pubKeys) == 1 {
```
Comment (crawford) [inline pkg/asset/installconfig/ssh.go]: We still need to allow the user to opt out of using a key at all.

Comment (wking) [reply pkg/asset/installconfig/ssh.go]: > We still need to allow the user to opt out of using a key at all.

I think we do.  Some cases:

* You set `OPENSHIFT_INSTALL_SSH_PUB_KEY` and exit [up here][1].
* You set `OPENSHIFT_INSTALL_SSH_PUB_KEY_PATH`, [get that single entry in your `pubKeys`][2], and exit here without a prompt.
* You set no variables, so you [get the `<none>` choice injected][3].  Then...
    * Go finds no valid pubkeys in `~/.ssh/*.pub`, and you exit here with the none choice selected.
    * Go finds some valid pubkeys in `~/.ssh/*.pub`, and you pass by here and fall down to the interactive prompt.

[1]: https://github.com/openshift/installer/pull/351/files#diff-0fa610c35b30a72dea1ff2a8c0b1e2e9R49
[2]: https://github.com/openshift/installer/pull/351/files#diff-0fa610c35b30a72dea1ff2a8c0b1e2e9R62
[3]: https://github.com/openshift/installer/pull/351/files#diff-0fa610c35b30a72dea1ff2a8c0b1e2e9R64

Comment (crawford) [reply pkg/asset/installconfig/ssh.go]: Ah, that seems okay then.

**Comment by crawford** [comment]
@wking If you rebase this, I'll LGTM it.

**Comment by wking** [comment]
> @wking If you rebase this, I'll LGTM it.

Does it need a rebase?

**Comment by crawford** [comment]
I just want to be sure that this has the changes that I requested back in #350. This should just be two commits.

**Comment by crawford** [comment]
Nevermind, I see the commit hash is the same.

/lgtm

**Comment by crawford** [comment]
On second thought, can we just drop `OPENSHIFT_INSTALL_SSH_PUB_KEY`? I don't want to have too many options here since we'll get support for reading an install config directly.

**Comment by wking** [comment]
> On second thought, can we just drop `OPENSHIFT_INSTALL_SSH_PUB_KEY`?

We could, I just don't like touching the disk if the caller already happens to have this in memory.  But having this in memory is unlikely (maybe for unit tests?), so I'm ok dropping it if you want.

> I don't want to have too many options here since we'll get support for reading an install config directly.

With some variables read directly and some read from paths, at least #352 is going to need to support both approaches.  The SSH reader could be simplified slightly if we only supported one approach, but I think it's going to be a difference of ~10 lines.  Is that too much?

**Comment by wking** [comment]
/retest

#352 was happy (before I had to push to fix `gofmt` ;), so the e2e errors must be flakes.

**Comment by crawford** [comment]
> The SSH reader could be simplified slightly if we only supported one approach, but I think it's going to be a difference of ~10 lines. Is that too much?

I'm not concerned with the amount of code so much as I'm concerned about supporting all of these interfaces.

**Comment by crawford** [comment]
In the interest of time, let's leave this as is. We can clean it up after CI is working.

**Pull request merged**